### ISSUE TITLE
Debugger local variables frozen after Go panic

### ISSUE THREAD

**Issue by nmeocisco** [Microsoft/vscode#72426]
When debugging a Go program, if a panic occurs then the local variables drop down will be frozen with a spinning circle displayed. This stays forever so vscode has to be closed and reopened for the problem to go away.

- VSCode Version: 1.33.1
- OS Version: Ubuntu 18.04
- Go extension: ms-vscode.go 0.9.2

Steps to Reproduce:

1. Create a simple test Go file that looks like the following:

```
package main

import (
	"fmt"
	"testing"
)

func TestExample(t *testing.T) {
	s := "example"
	panic(s)
}

func TestExample2(t *testing.T) {
	fmt.Println("example")
}
```

2. Add a breakpoint on the line containing the "panic" call as shown in the following screenshot.

![image](https://user-images.githubusercontent.com/49691353/56228319-8876ae80-6045-11e9-9722-cb0f269886df.png)

3. Click on "debug test" above TestExample() to start debugging.
4. The debugger will stop at panic. Notice so far everything looks good and you can see the value of "s" shown in the local variables on the left side.

![image](https://user-images.githubusercontent.com/49691353/56228414-ca075980-6045-11e9-814d-958998b5c395.png)

5. Hit continue to allow the panic to occur. Notice that there is now a spinning circle next to "Local" and it is still displaying the value of "s" even though it is out of scope.

![image](https://user-images.githubusercontent.com/49691353/56228920-c1635300-6046-11e9-9ea6-fb8ef48f624b.png)

6. Allow the program to continue and exit. Now start debugging TestExample2() and you will notice the local variables display is stuck. In the following screenshot it still shows a value for "s" even though TestExample2() does not have it defined.

![image](https://user-images.githubusercontent.com/49691353/56229081-1b641880-6047-11e9-82ef-2e4be990fc6d.png)

**Comment by lggomez** [comment]
@isidorn this is happening across two different debug sessions, shouldn't the watch and variables panes be cleaned up in between them? even if a variables request hangs, my understanding is that once that debug session is terminated the debug adapter process is killed and there shouldn't be any further dependencies active outside the main workbench process holding this data

**Comment by isidorn** [comment]
@lggomez yeah in a perfect world we would cancel the requests which do not come back after some time. I think there is even an open issue for this and we might imporve it in the future.

**Comment by lggomez** [comment]
@isidorn thanks for the clarification. Do you know if there is a way to do a cleanup from the extension side or, for the time being, is the debug adapter responsible for handling any anomalous response from the debugger?
